params/json.libevm.go

```diff
@@ -124,3 +124,27 @@ func toJSONRawMessages(v any) (map[string]json.RawMessage, error) {
 	}
 	return msgs, nil
 }
+
+// UnmarshalChainConfig JSON decodes the given data into the config and the extra
+// pointer arguments. The root JSON keys except the "extra" key are decoded into
+// the config argument, and the object at the "extra" key, if present, is decoded
+// into the extra argument. Note the extra argument must be a non-nil pointer or
+// an error would be returned.
+func UnmarshalChainConfig(data []byte, config *ChainConfig, extra any) (err error) {
+	err = json.Unmarshal(data, config)
+	if err != nil {
+		return fmt.Errorf("json decoding root chain config: %w", err)
```

Comment:
(informational) I know that you have a strong preference towards (1) annotating errors and (2) also always using `%w`. I'd like to be a little more nuanced than that in this code base. My rationale for each, respectively:

1. I'm ok with some over-annotation because it's better than insufficient, but keep in mind what the annotated error will have and what the recipient will do. In this case I think it's important to differentiate between the root and the extra so I'm pro the annotation. I really like that your annotations only provide context and not "error when" or "failed to" because a bunch of those when bubbling the error are cluttered.

2. `%w` should be a motivated choice instead of the default because it makes the wrapped error part of the public interface; sometimes that's beneficial (e.g. usually `io.EOF` or `fs.ErrNotExist` tend to be) and sometimes it's at risk of [Hyrum's Law](https://www.hyrumslaw.com/). One heuristic I use is whether the wrapped error is programmatically useful (e.g. can it be `==` or `errors.Is` compared), which I don't think any of the `json` package errors are.

Reply:
> not "error when" or "failed to" because a bunch of those when bubbling the error are cluttered.

Don't tell me this, I feel like I found a clone of myself 🤣  I kept on telling everyone coding in Go about this, first time I meet someone knowing about this cluttering! 👍 💯 



> 2\. sometimes that's beneficial (e.g. usually `io.EOF` or `fs.ErrNotExist` tend to be) and sometimes it's at risk of [Hyrum's Law](https://www.hyrumslaw.com/).

Agreed as well, I got used to just wrap everything by default, maybe a pretty bad habit to be fair. They even say it on the [Go 1.13 error notes](https://go.dev/blog/go1.13-errors#:~:text=Is(err%2C%20ErrPermission)%20...-,Whether%20to%20Wrap%C2%B6,-When%20adding%20additional) 

> Wrap an error to expose it to callers. Do not wrap an error when doing so would expose implementation details.

And it's true most of time we just want to log it out as a string! I'll re-orient to using `%s` instead by default 👍